File: 2019/2019W34/preppindataw34.R
library(readxl)
library(dplyr)
library(stringr)
library(lubridate)

Sys.setlocale("LC_ALL","English")

final <- read_excel(path = "E:/PD - Week 34.xlsx", 
                    sheet = "Delivery Schedule") %>%
  mutate('nth' = as.integer(str_extract(`Delivery Schedule`,'^(\\d+)')),
         'Weekday' = str_extract(`Delivery Schedule`,'(?<=\\s).*(?=\\sof)')) %>%
  merge(., read_excel(path = "E:/PD - Week 34.xlsx", 
                      sheet = "Date Scaffold") %>%
          mutate('Month Name' = strftime(`Date`,'%B'),
                 'YearMonth' = strftime(`Date`,'%Y%m'),
                 'Weekday' = strftime(`Date`,'%A'),
                 'one' = 1) %>%
          group_by(`YearMonth`,`Weekday`) %>%
          arrange(`Date`) %>%
          mutate('nth' = cumsum(`one`)),
        on = c('Weekday','nth')) %>%
  arrange(`Date`) %>%
  select(c('Month Name','Weekday','Date','Product','Scent','Supplier','Quantity'))

View(final)  

File: 2019/2019W34/PreppindataW34.py
import pandas as pd
import re

sch = pd.read_excel(r'E:\PD - Week 34.xlsx','Delivery Schedule')
sch['nth'] = sch['Delivery Schedule'].apply(lambda x: int(re.search(r'^(\d+)',x).group(1)))
sch['Weekday'] = sch['Delivery Schedule'].apply(lambda x: re.search(r'.*\s(.*)\sof',x).group(1))

date = pd.read_excel(r'E:\PD - Week 34.xlsx','Date Scaffold')
date['Date'] = date['Date'].apply(lambda x: x.date())
date['YearMonth'] = date['Date'].apply(lambda x: x.strftime('%Y%m'))
date['Weekday'] = date['Date'].apply(lambda x: x.strftime('%A'))
date['Month Name'] = date['Date'].apply(lambda x: x.strftime('%B'))
date['nth'] = date.sort_values('Date').groupby(['YearMonth','Weekday']).cumcount()+1
date=date[['Date','Weekday','nth','Month Name']].copy()

final = pd.merge(sch,date,on=['Weekday','nth'],how='inner')
final=final[['Month Name','Weekday','Date','Product','Scent','Supplier','Quantity']].sort_values(by=['Date']).copy().reset_index(drop=True)
